packages/mobile/src/identity/contactMapping.ts

```diff
@@ -261,36 +258,43 @@ function* fetchWalletAddresses(e164Number: string) {
   return possibleUserAddresses
 }
 
-// Returns IdentifierLookupResult
-// which is Map of identifier -> (Map of address -> AttestationStat)
-export function* lookupAttestationIdentifiers(ids: string[]) {
+// Returns a list of accounts for the identifier received.
+export function* lookupAttestationIdentifiers(id: string) {
   const contractKit = yield call(getContractKit)
   const attestationsWrapper: AttestationsWrapper = yield call([
     contractKit.contracts,
     contractKit.contracts.getAttestations,
   ])
 
-  return yield call([attestationsWrapper, attestationsWrapper.lookupIdentifiers], ids)
+  return yield call([attestationsWrapper, attestationsWrapper.lookupAccountsForIdentifier], id)
 }
 
 // Deconstruct the lookup result and return
 // any addresess that are considered verified
-export function getAddressesFromLookupResult(
-  lookupResult: IdentifierLookupResult,
-  phoneHash: string
-) {
-  if (!lookupResult || !lookupResult[phoneHash]) {
-    return null
+export function* getAddressesFromLookupResult(lookupResult: string[], phoneHash: string) {
+  if (!lookupResult) {
+    return []
   }
 
-  const addressToStats = lookupResult[phoneHash]!
+  const contractKit = yield call(getContractKit)
+  const attestationsWrapper: AttestationsWrapper = yield call([
+    contractKit.contracts,
+    contractKit.contracts.getAttestations,
+  ])
+
   const verifiedAddresses: string[] = []
-  for (const address of Object.keys(addressToStats)) {
+  for (const address of lookupResult) {
     if (!isValidNon0Address(address)) {
       continue
     }
+    // Get stats for the address
+    const stats: AttestationStat = yield call(
```

Comment:
Doing a call for each account received. This could be an issue when called from `verifyIdentityMetadata` on `packages/mobile/src/identity/commentEncryption.ts`, but my understanding is that it's not run often, so I'm unsure of what the performance hit will actually be. Any thoughts on this? Should I call another batch method to get stats to better cover that case?

Reply:
If a batched request can be made to the contract, I think it's worth it. Otherwise, you can break this loop up to make it more performant by:
- filtering invalid addresses
- doing a `yield all` with a `call` for the attestation state of every valid address
- filtering for verified addresses